burstphoto/exposure/exposure.swift

```diff
@@ -28,68 +28,77 @@ func correct_exposure(_ final_texture: MTLTexture, _ white_level: Int, _ black_l
             }
         }
         
-        var black_level0 = 0.0
-        var black_level1 = 0.0
-        var black_level2 = 0.0
-        var black_level3 = 0.0
-        
+        var black_levels_mean = Array(black_level[exp_idx].map{Double($0)})
+
         // if exposure levels are uniform, calculate mean value of all exposures
         if uniform_exposure {
-            
-            for comp_idx in 0..<exposure_bias.count {
-                black_level0 += Double(black_level[comp_idx][0])
-                black_level1 += Double(black_level[comp_idx][1])
-                black_level2 += Double(black_level[comp_idx][2])
-                black_level3 += Double(black_level[comp_idx][3])
+            for i in 0..<black_level.count {
+                for j in 0..<black_levels_mean.count {
+                    black_levels_mean[j] += Double(black_level[i][j])
```

Comment:
I believe the values in the vector black_levels_mean have to be set to zero before the summation begins. Otherwise the index exp_idx will be added twice.